### PR TITLE
Decouple frees config

### LINKED ISSUE #294: Decouple from frees-config
we're depending on frees-config for the config module (on which others depend). We need to stop depending on it for #290

Related to https://github.com/frees-io/freestyle-rpc/issues/290

### PR DESCRIPTION
**Pull request by pepegar**
In this PR I:

- create a bare minimum `ConfigM` tagless to be used wherever `freestyle.tagless.config` were used before
- remove a couple of `@module` annotations
- bump `frees-todolist-lib`to `v0.8.1`
- bump sbt-freestyle to `0.3.23`

FIXES #294

### CODE REVIEW

File: modules/config/src/main/scala/client/ChannelConfig.scala [hunk 1]
```diff
@@ -18,27 +18,46 @@ package freestyle.rpc
...
+      target <- sync.pure(Either.catchOnly[Missing](config.getString(targetPath)))
+    } yield ChannelForTarget(target.getOrElse("target"))
+
+}
+
+object ChannelConfig {
+  def apply[F[_]](implicit S: Sync[F], C: ConfigM[F]): ChannelConfig[F] = new ChannelConfig[F] {
+    def sync    = S
+    def configM = C
+  }
```
Comment (juanpedromoreno) [inline modules/config/src/main/scala/client/ChannelConfig.scala]: I'd suggest the following approach, more natural perhaps?

```scala
class ChannelConfig[F[_]](implicit S: Sync[F], C: ConfigM[F]) {

  val defaultHost: String = "localhost"
  val defaultPort: Int    = freestyle.rpc.server.defaultPort

  def loadChannelAddress(hostPath: String, portPath: String): F[ChannelForAddress] =
    for {
      config <- C.load
      host <- S.pure(
        Either
          .catchOnly[Missing](config.getString(hostPath)))
      port <- S.pure(Either.catchOnly[Missing](config.getInt(portPath)))
    } yield ChannelForAddress(host.getOrElse(defaultHost), port.getOrElse(defaultPort))

  def loadChannelTarget(targetPath: String): F[ChannelForTarget] =
    for {
      config <- C.load
      target <- S.pure(Either.catchOnly[Missing](config.getString(targetPath)))
    } yield ChannelForTarget(target.getOrElse("target"))

}

object ChannelConfig {
  def apply[F[_]](implicit S: Sync[F], C: ConfigM[F]): ChannelConfig[F] = new ChannelConfig[F]

  implicit def defaultChannelConfig[F[_]](implicit S: Sync[F], C: ConfigM[F]): ChannelConfig[F] =
    apply[F]
}
```

File: modules/config/src/main/scala/server/ServerConfig.scala [hunk 2]
```diff
@@ -18,26 +18,44 @@ package freestyle.rpc
...
+
+import freestyle.rpc.config.ConfigM
+
+import com.typesafe.config.ConfigException.Missing
 
-@module
-trait ServerConfig {
+trait ServerConfig[F[_]] {
+
+  implicit def sync: Sync[F]
```
Comment (juanpedromoreno) [inline modules/config/src/main/scala/server/ServerConfig.scala]: Same here

File: modules/config/src/main/scala/config.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import cats.effect.Sync
+import pureconfig._
+import com.typesafe.config.{Config, ConfigFactory}
+
+object config {
+
+  trait ConfigM[F[_]] {
+    def load: F[Config]
+  }
```
Comment (fedefernandez) [inline modules/config/src/main/scala/config.scala]: I'm wondering if we could add a new method to this algebra:
```scala
def get[A](from: (Config) => A): F[Either[Missing, A]]
```
With the implementation:
```scala
def get[A](getValue: => A): F[Either[Missing, A]] =
  S.pure(Either.catchOnly[Missing](getValue))
```

Comment (pepegar) [reply modules/config/src/main/scala/config.scala]: I think that, at some point in the near future, we'll reintroduce `frees-config`, and we could maybe add it there? this implementation is temporary. What do you think?

**Pull request merged**